README.md

```diff
@@ -71,7 +71,14 @@ module.exports = binding
 An added benefit of this approach is that your native modules will work across multiple node and electron versions without having the user
 need to reinstall or recompile them - as long as you produce prebuilds for all versions. With Node-API you only have to produce prebuilds for every runtime.
 
-When publishing your module to npm remember to include the `./prebuilds` folder.
+
+## Platform-specific Packages
+As an alternate to including all prebuilds directly in your published package, you can use  `--optional-packages` to setup the prebuilds for publishing as separate platform-specific packages. Using this option, each prebuild directory will also include a package.json that specifies the target platform and architectures along with some basic package files. Each of these prebuild directories can then be published as separate packages in NPM. In addition,
+the main package should specify all the platform packages as `optionalDependencies`. When installed, NPM (or Yarn, etc.) will then only install the optional dependency with the platform/architecture matching the target machine.
+
+This provides both the efficiency of only needing to download/install the binary (or binaries) needed for the current platform, but also the key benefit of the prebuildify system in that binaries are downloaded and available as part of the normal NPM install process (no install scripts for extra downloads are necessary). When using this method, you should omit the prebuilds folder when publishing your package (since they will be separately downloaded).
```

Comment:
```suggestion
This provides both the efficiency of only needing to download the binaries needed for the current platform, and the key benefit of the `prebuildify` approach in that binaries are downloaded as part of the normal npm install process (without install scripts). When using this option, you should omit the `./prebuilds` folder when publishing your main package, since they will be separately downloaded by npm.
```

Reply:
One thing to be aware of is that this doesn't universally avoid downloads of unnecessary platform packages (yet), but does universally avoid extra installs:
yarn & npm 6 - Will still download all platform packages to the global package cache, but will only install relevant one.
npm 7+ - Will only download package meta data and only fully download the correct platform package (and install it).

I could update the docs to clarify that.

BTW, if you want to see download frequencies, I actually have a few stats now, since I have been using these optional/platform packages for a few weeks now on a couple of my packages:
[linux arm 32 bit build](https://www.npmjs.com/package/msgpackr-extract-linux-arm) - 127K/week, I think almost no one has this platform, so this represents baseline downloads of users that use a package manager that downloads all platforms.
[linux x64](https://www.npmjs.com/package/msgpackr-extract-linux-x64) - 197K/week, I think this is the most popular platform (a lot of CI servers doing these downloads), so probably about 60K/week downloads from package managers that are only downloading the correct platform
[win32](https://www.npmjs.com/package/msgpackr-extract-win32-x64) - 132K/week, another platforms for comparison.

Reply:
> I could update the docs to clarify that.

That would be good